test/provisioner/provision_azure.go

```diff
@@ -8,37 +8,351 @@ package provisioner
 import (
 	"context"
 
+	"errors"
+	"fmt"
+	"os"
 	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/containerservice/mgmt/containerservice"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/network/mgmt/network"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/resources/mgmt/resources"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/to"
+	log "github.com/sirupsen/logrus"
+	kconf "sigs.k8s.io/e2e-framework/klient/conf"
 )
 
 func init() {
 	newProvisionerFunctions["azure"] = NewAzureCloudProvisioner
 }
 
-// AzureCloudProvisioner implements the CloudProvisioner interface for Azure.
+func CreateResourceGroup() error {
+	newRG := resources.Group{
+		Location: &AzureProps.Location,
+	}
+
+	log.Infof("Creating Resource group %s.\n", AzureProps.ResourceGroupName)
+	resourceGroup, err := AzureProps.ResourceGroupClient.CreateOrUpdate(context.Background(), AzureProps.ResourceGroupName, newRG)
+	if err != nil {
+		fmt.Println("Failed to create resource group:", err)
+		return err
+	}
+
+	AzureProps.ResourceGroup = &resourceGroup
+
+	log.Infof("Successfully Created Resource group %s.\n", AzureProps.ResourceGroupName)
+	return nil
+}
+
+func DeleteResourceGroup() error {
+	log.Infof("Deleting Resource group %s.\n", AzureProps.ResourceGroupName)
+	_, err := AzureProps.ResourceGroupClient.Delete(context.Background(), AzureProps.ResourceGroupName)
+	if err != nil {
+		fmt.Println("Failed to delete resource group:", err)
+		return err
+	}
+
+	log.Infof("Successfully Deleted Resource group %s.\n", AzureProps.ResourceGroupName)
+
+	return nil
+}
+
+func CreateVnetSubnet() error {
+	addressPrefix := "10.2.0.0/16"
+	subnetAddressPrefix := "10.2.0.0/24"
+	vnetParams := network.VirtualNetwork{
+		Location: &AzureProps.Location,
+		Name:     &AzureProps.VnetName,
+		VirtualNetworkPropertiesFormat: &network.VirtualNetworkPropertiesFormat{
+			AddressSpace: &network.AddressSpace{
+				AddressPrefixes: &[]string{addressPrefix},
+			},
+			Subnets: &[]network.Subnet{
+				{
+					Name: &AzureProps.SubnetName,
+					SubnetPropertiesFormat: &network.SubnetPropertiesFormat{
+						AddressPrefix: &subnetAddressPrefix,
+					},
+				},
+			},
+		},
+	}
+
+	// Create the virtual network
+	log.Infof("Creating  vnet %s in resource group %s with addressPrefix %s subnetAddressPrefix %s.\n", AzureProps.VnetName, AzureProps.ResourceGroupName, addressPrefix, subnetAddressPrefix)
+	_, err := AzureProps.ManagedvnetClient.CreateOrUpdate(context.Background(), AzureProps.ResourceGroupName, AzureProps.VnetName, vnetParams)
```

Comment:
I guess AzureProps.ManagedvnetClient.CreateOrUpdate will not fail if vnet with same name already exists?

Reply:
Yes it wont fail even if it already exist